### PR TITLE
HDDS-2720. Ozone Failure injection Service

### PR DESCRIPTION
**Pull request by prashantpogde**
## What changes were proposed in this pull request?

Adding Fault Injection Service in Ozone:tools/FaultInjectionService directory

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2720

## How was this patch tested?

This is an independent tool which will be used to do fault injection testing for Ozone. Its build/test process is independent of Ozone and should not impact Ozone in any way.
For using this tool, please follow README in tools/FaultInjectionService.

### CODE REVIEW

File: tools/FaultInjectionService/AUTHORS [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Current Maintainer
```
Comment (elek) [inline tools/FaultInjectionService/AUTHORS:1]: I am not sure if we need this file (but I can be convinced). Until now we didn't follow this pattern. There are no maintainers of SCM / Freon / OM. The community itself is the maintainers but we can check the git history for first contact.

This file is also requires additional maintenance. I would suggest to remove it.

File: tools/FaultInjectionService/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+NoiseInjector
+==============
+
+About
+------
+TBD
```
Comment (elek) [inline tools/FaultInjectionService/README.md:6]: Just one sentence please

File: tools/FaultInjectionService/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Development Status                                                              
+------------------
+TBD
```
Comment (elek) [inline tools/FaultInjectionService/README.md:10]: One word, please ;-)

File: tools/FaultInjectionService/AUTHORS (hunk 1 above)
Comment (prashantpogde) [reply tools/FaultInjectionService/AUTHORS:1]: @elek Thank you for reviewing it. 
- Yes I will remove AUTHORS file and change the directory name to lower case.
- Its supported only on linux and some dependencies require building them from the source. I will add an automated build script as next set of enhancement and create a separate PR.

File: tools/FaultInjectionService/README.md (hunk 2 above)
Comment (prashantpogde) [reply tools/FaultInjectionService/README.md:6]: yup, will change.

File: tools/FaultInjectionService/README.md (hunk 3 above)
Comment (prashantpogde) [reply tools/FaultInjectionService/README.md:10]: yup, will change.

**Comment by elek** [comment]
Thanks the update @prashantpogde 

 1. Can you please remove `tools/fault-injection-service/LICENSE`? As I wrote this is unnecessary as the whole project is distributed under the top level `LICENSE` file.

 2. License header check doesn't work for this dir (as I see). Can you please double check if ASF headers are added to everywhere? Seems to be missing from the `Test*.cc` files 

 3. I am fine to create the build script in the next PR. After 1-2, I will merge this.

**Comment by prashantpogde** [comment]
@elek Thank you for the review. I have made the suggested changes. Please take a look.

File: tools/fault-injection-service/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+Fault Injection Service
```
[the remark this answers is not in the document]
Comment (prashantpogde) [reply tools/fault-injection-service/README.md:1]: I noticed that ASF license header was not there in Ozone README.md,  Should I put it in this one ?

**Comment by elek** [comment]
I just found this one on the [faq page](https://apache.org/legal/src-headers.html)

> Other files may make sense to have no license header. Three examples are: ... Short informational text files; for example README, INSTALL files

It seems that we don't need LICENSE header for that specific README.

**Pull request merged**